tests/e2e/common/parameterized_query_tests.py

```diff
@@ -34,8 +37,19 @@ def _quantize(self, input: Union[float, int], place_value=2) -> Decimal:
 
         return Decimal(str(input)).quantize(Decimal("0." + "0" * place_value))
 
-    def test_primitive_inferred_none(self):
+    @patch(
+        "databricks.sql.client.Connection.server_parameterized_queries_enabled",
+        return_value=False,
+    )
+    def test_protocol_too_low(self, mock_parameterized_queries_enabled):
+        params = {"p": None}
+        with pytest.raises(
+            Error,
```

Comment:
Might as well make this test more specific. A `NotSupportedError` should be raised. Else this test will pass for _any_ known exception (connection failed, syntax error, network stack failure etc)